### DISCUSSION TITLE
Auto-closing Dialogs upon submission

### DISCUSSION THREAD

**Discussion by nthskyradiated** [huntabyte/shadcn-svelte#1097] [Help]
Hi Hunter / Everyone,

Is there a way to auto-close dialogs upon submission / confirmation?

Right now I find they only close with the close button or by clicking outside them.

Thanks for the great work!

**Comment by shyakadavis** [comment]
Hey, @nthskyradiated 

Without enough context, I'm just speculating, but I believe binding the `open` prop on `Root` to a state variable, and just setting it to false after submitting should do the trick.

```svelte
<script>
let open = false;

function submit_form(){
// do something
open = false;
}
</script>

<Dialog.Root bind:open>
</Dialog.Root>
```

**Reply by nthskyradiated** [reply]
thanks!
